nixos/lib/testing/nodes.nix

```diff
@@ -27,13 +46,15 @@ let
           ({ config, ... }:
             {
               virtualisation.qemu.package = testModuleArgs.config.qemu.package;
+              virtualisation.host.pkgs = hostPkgs;
+              nixpkgs.pkgs = guestPkgs;
             })
           ({ options, ... }: {
             key = "nodes.nix-pkgs";
             config = optionalAttrs (!config.node.pkgsReadOnly) (
               mkIf (!options.nixpkgs.pkgs.isDefined) {
                 # TODO: switch to nixpkgs.hostPlatform and make sure containers-imperative test still evaluates.
-                nixpkgs.system = hostPkgs.stdenv.hostPlatform.system;
+                nixpkgs.system = hostPlatform.system;
```

Comment:
This should probably be the guest system.

Note that this definition only applies when the `pkgs` is not read only - ie we need to construct a new Nixpkgs.
With that in mind, we should not evaluate `pkgsNative.${guestSystem}`, as that Nixpkgs instance will usually not be used.

By improving this definition to use the guest system, we can support darwin as hypervisor for NixOS tests that have overlays, etc.